### PR TITLE
fix(request-transformer) an empty JSON array after being decoded, can…

### PR DESCRIPTION
**Pull request by catbro666**
…'t be properly re-encoded as an array. Instead, it becomes an empty json object.

This PR fixes this issue. An empty json array will remain to be an empty json array after being decoded and re-encoded.

This is a sister PR to PR:

[#4901](https://github.com/Kong/kong/pull/4901)

It contains a similar fix, but now to `request-transformer` (this plugin).

FTI-4205

### CODE REVIEW

**Review by ms2008 — APPROVED** [review]
LGTM

**Review by chronolaw — APPROVED** [review]

File: kong/plugins/request-transformer/access.lua [hunk 1]
```diff
@@ -1,5 +1,5 @@
 local multipart = require "multipart"
-local cjson = require "cjson"
+local cjson = require("cjson.safe").new()
```
Comment (flrgh) [inline kong/plugins/request-transformer/access.lua:2]: If you're going to switch from `cjson` to `cjson.safe` then you need to update the `parse_json` local function below, as it uses `pcall`.

**Review by flrgh — CHANGES_REQUESTED** [review]

**Review by ms2008 — CHANGES_REQUESTED** [review]
https://github.com/Kong/kong/blob/a9e58a1f3c8bd2b89b1a7be67c9179d15bd06147/kong/plugins/request-transformer/access.lua#L22

This line also needs to be removed.

**Review by flrgh — APPROVED** [review]
:+1: lgtm; good to go after removing the now-unused `local pcall` as per @ms2008's comment.

**Comment by catbro666** [comment]
> 

done

**Review by ms2008 — APPROVED** [review]

**Pull request merged**